src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -1310,54 +1369,94 @@ public int CompareTo(object? target)
+
                 case CorElementType.ELEMENT_TYPE_U1:
                 case CorElementType.ELEMENT_TYPE_BOOLEAN:
                     return pThisValue.CompareTo(pTargetValue);
+
                 case CorElementType.ELEMENT_TYPE_I2:
                     return Unsafe.As<byte, short>(ref pThisValue).CompareTo(Unsafe.As<byte, short>(ref pTargetValue));
+
                 case CorElementType.ELEMENT_TYPE_U2:
                 case CorElementType.ELEMENT_TYPE_CHAR:
                     return Unsafe.As<byte, ushort>(ref pThisValue).CompareTo(Unsafe.As<byte, ushort>(ref pTargetValue));
+
                 case CorElementType.ELEMENT_TYPE_I4:
 #if TARGET_32BIT
                 case CorElementType.ELEMENT_TYPE_I:
 #endif
                     return Unsafe.As<byte, int>(ref pThisValue).CompareTo(Unsafe.As<byte, int>(ref pTargetValue));
+
                 case CorElementType.ELEMENT_TYPE_U4:
 #if TARGET_32BIT
                 case CorElementType.ELEMENT_TYPE_U:
 #endif
                     return Unsafe.As<byte, uint>(ref pThisValue).CompareTo(Unsafe.As<byte, uint>(ref pTargetValue));
+
                 case CorElementType.ELEMENT_TYPE_I8:
 #if TARGET_64BIT
                 case CorElementType.ELEMENT_TYPE_I:
 #endif
                     return Unsafe.As<byte, long>(ref pThisValue).CompareTo(Unsafe.As<byte, long>(ref pTargetValue));
+
                 case CorElementType.ELEMENT_TYPE_U8:
 #if TARGET_64BIT
                 case CorElementType.ELEMENT_TYPE_U:
 #endif
                     return Unsafe.As<byte, ulong>(ref pThisValue).CompareTo(Unsafe.As<byte, ulong>(ref pTargetValue));
+
                 case CorElementType.ELEMENT_TYPE_R4:
                     return Unsafe.As<byte, float>(ref pThisValue).CompareTo(Unsafe.As<byte, float>(ref pTargetValue));
+
                 case CorElementType.ELEMENT_TYPE_R8:
                     return Unsafe.As<byte, double>(ref pThisValue).CompareTo(Unsafe.As<byte, double>(ref pTargetValue));
+
                 default:
                     Debug.Fail("Unknown enum underlying type");
                     return 0;
             }
         }
-        #endregion
 
-        #region IFormattable
-        [Obsolete("The provider argument is not used. Use ToString(String) instead.")]
-        public string ToString([StringSyntax(StringSyntaxAttribute.EnumFormat)] string? format, IFormatProvider? provider)
+        /// <summary>Converts the value of this instance to its equivalent string representation.</summary>
+        /// <remarks>The string representation of the value of this instance.</remarks>
+        public override string ToString()
         {
-            return ToString(format);
+            RuntimeType enumType = (RuntimeType)GetType();
+            ref byte rawData = ref this.GetRawData();
+            CorElementType corElementType = InternalGetCorElementType();
+            return corElementType switch
```

Comment:
I wonder how would GetTypeCode on enumType compare here, if the method gets inlined it could then be turned into a constant and the type code cache was quite fast last time I've checked.

Reply:
Using the type code measured worse.